### PR TITLE
Add support for BSSID configuration in connect call.

### PR DESCRIPTION
**Pull request by rado17**
test_wifi: sdk-nrf-pr-14626

### CODE REVIEW

**Review by sachinthegreen — APPROVED** [review]

File: modules/hostap/src/supp_api.c [hunk 1]
```diff
@@ -398,6 +398,11 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 		}
 	}
 
+	if (params->bssid[0] != 0) {
```
Comment (krish2718) [inline modules/hostap/src/supp_api.c:401]: `....00` this is still a valid BSSID, no?

Comment (rado17) [reply modules/hostap/src/supp_api.c:401]: It is. But here, we are checking this against NULL character (ASCII value = 0).

Comment (krish2718) [reply modules/hostap/src/supp_api.c:401]: Yes, so, a BSSID like `00:11:22:33:44:00` would be considered NULL and hence not valid. We should check for all zeroes  to handle the case where user hasn't set or not. And if non-zero, then do basic validation and only then accept.

**Review by VivekUppunda — APPROVED** [review]

**Review by kapbh — APPROVED** [review]

File: modules/hostap/src/supp_api.c [hunk 2]
```diff
@@ -403,6 +404,19 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 		}
 	}
 
+	memcpy((void *)&mac, params->bssid, WIFI_MAC_ADDR_LEN);
+	if (!net_eth_is_addr_broadcast(&mac) &&
```
Comment (krish2718) [inline modules/hostap/src/supp_api.c:408]: we should fail for invalid BSSID except for all zeroes (which can be treated as unset).

File: modules/hostap/src/supp_api.c [hunk 3]
```diff
@@ -403,6 +404,25 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 		}
 	}
 
+	memcpy((void *)&mac, params->bssid, WIFI_MAC_ADDR_LEN);
+	if (net_eth_is_addr_broadcast(&mac) ||
```
Comment (krish2718) [inline modules/hostap/src/supp_api.c:408]: You move this check in to the below branch (!unspecified)

**Review by krish2718 — APPROVED** [review]

File: modules/hostap/src/supp_api.c (hunk 3 above)
Comment (rado17) [reply modules/hostap/src/supp_api.c:408]: !unspecified checks for all 0s which is equivalent to `not setting the BSSID`. That is why I am returning error only for broadcast and multicast case.

File: modules/hostap/src/supp_api.c (hunk 2 above)
Comment (rado17) [reply modules/hostap/src/supp_api.c]: Done

**Review by jukkar — APPROVED** [review]

**Comment by rlubos** [comment]
@rado17 Needs a rebase.

**Pull request merged**